ca_description/urdf/sensors/cliff_sensors_gazebo.xacro

```diff
@@ -7,118 +7,43 @@
-      <pose>0 0 0 0 0 0</pose>
-      <visualize>${visualize}</visualize>
-      <ray>
-        <scan>
-          <horizontal>
-            <samples>1</samples>
-            <resolution>1</resolution>
-            <min_angle>0</min_angle>
-            <max_angle>0</max_angle>
-          </horizontal>
-        </scan>
-        <range>
-          <min>${min_range_m}</min>
-          <max>${max_range_m}</max>
-          <resolution>0.1</resolution>
-        </range>
-      </ray>
-      <plugin name="gazebo_ros_${name_front_left}_cliff_sensor" filename="libgazebo_ros_laser.so">
-        <topicName>${name_front_left}_cliff_sensor/scan</topicName>
-        <frameName>${name_front_left}_cliff_sensor</frameName>
-      </plugin>
-    </sensor>
-  </gazebo>
-
-  <gazebo reference="${name_front_right}_cliff_sensor_link">
-    <sensor type="ray" name="${name_front_right}_cliff_sensor">
-      <always_on>true</always_on>
-      <update_rate>${rate_hz}</update_rate>
-      <pose>0 0 0 0 0 0</pose>
-      <visualize>${visualize}</visualize>
-      <ray>
-        <scan>
-          <horizontal>
-            <samples>1</samples>
-            <resolution>1</resolution>
-            <min_angle>0</min_angle>
-            <max_angle>0</max_angle>
-          </horizontal>
-        </scan>
-        <range>
-          <min>${min_range_m}</min>
-          <max>${max_range_m}</max>
-          <resolution>0.1</resolution>
-        </range>
-      </ray>
-      <plugin name="gazebo_ros_${name_front_right}_cliff_sensor" filename="libgazebo_ros_laser.so">
-        <topicName>${name_front_right}_cliff_sensor/scan</topicName>
-        <frameName>${name_front_right}_cliff_sensor</frameName>
-      </plugin>
-    </sensor>
-  </gazebo>
+  <xacro:property name="min_cliff_value_cm" value="3.5"/>
+
+  <xacro:macro name="gazebo_cliff_sensor" params="name">
+    <gazebo reference="${name}_cliff_sensor_link">
+      <sensor type="ray" name="${name}_cliff_sensor">
+        <always_on>true</always_on>
+        <update_rate>${rate_hz}</update_rate>
+        <pose>0 0 0 0 0 0</pose>
+        <visualize>${visualize}</visualize>
+        <ray>
+          <scan>
+            <horizontal>
+              <samples>1</samples>
+              <resolution>1</resolution>
+              <min_angle>0</min_angle>
+              <max_angle>0</max_angle>
+            </horizontal>
+          </scan>
+          <range>
+            <min>${min_range_m}</min>
+            <max>${max_range_m}</max>
+            <resolution>0.1</resolution>
+          </range>
+        </ray>
+        <plugin name="${name}_cliff_sensor" filename="libcreate_cliff_plugin.so">
+          <topicName>${name}_cliff_sensor/raw</topicName>
+          <frameName>${name}_cliff_sensor_link</frameName>
+          <cliffValue>${min_cliff_value_cm*cm_to_meters}</cliffValue>
+        </plugin>
+      </sensor>
+    </gazebo>
+  </xacro:macro>
+
+  <xacro:gazebo_cliff_sensor name="${name_side_left}"/>
```

Comment:
I didn't change the name that the old Cliff plugin was using, I can change it.

Reply:
Is it ok to change to pass the name as a variable here? because naming the argument the same name as its value seems redundant, not sure how to proceed.

Reply:
You'll need to pass the name as an argument, or do you have another idea?